src/main/java/net/neoforged/neoforge/event/BuildCreativeModeTabContentsEvent.java

```diff
@@ -61,12 +64,127 @@ public boolean hasPermissions() {
         return this.parameters.hasPermissions();
     }
 
-    public MutableHashedLinkedMap<ItemStack, CreativeModeTab.TabVisibility> getEntries() {
-        return this.entries;
+    /**
+     * The current immutable list of the parent tab entries in the order to be added to the Creative Menu.
+     * Purely for querying to see what in it. Please use the other event methods for modifications.
+     */
+    public List<ItemStack> getParentEntries() {
+        return this.parentEntries.stream().toList();
+    }
+
+    /**
+     * The current immutable list of the search tab entries in the order to be added to the Creative Menu.
+     * Purely for querying to see what in it. Please use the other event methods for modifications.
+     */
+    public List<ItemStack> getSearchEntries() {
+        return this.searchEntries.stream().toList();
     }
```

Comment:
Why are these lists instead of immutable SortedSets? Is it expected that users would want to query based on index?

Reply:
It's a list so users can see what order the itemstacks are in as if I return a set, it's not clear if it is in a specific order and a bit harder to check where stuff is in it imo. Immutable too to force users to use the other methods to mutate the underlying ordered set we use that is more of an implementation detail we hide away from end modders.

Reply:
Sorry if what I wrote was confusing, I'll restate it differently. Why can't these methods return an immutable `SortedSet`, `ObjectSortedSets.unmodifiable(this.parentEntries)` so that it's just a view of the implementation rather than a full copy? I know that you don't want to return a `Set` by itself because that interface does not imply any order, though `SortedSet` does (albeit, the interface implies either Comparator ordering or natural ordering, which this impl does not obey).

Reply:
Ah. Yeah I can return that. Users could do a toList themselves if they really need index but can't think of a specific reason why they would need index and not just use putAfter and have us handle finding location for them